include/grpc/support/port_platform.h

```diff
@@ -611,6 +588,35 @@ typedef unsigned __int64 uint64_t;
 #endif
 #endif /* GPR_HAS_CPP_ATTRIBUTE */
 
+#if defined(__GNUC__) && !defined(__MINGW32__)
+#define GPR_ALIGN_STRUCT(n) __attribute__((aligned(n)))
+#else
+#define GPR_ALIGN_STRUCT(n)
+#endif
+
+#ifndef GRPC_MUST_USE_RESULT
+#if GPR_HAS_CPP_ATTRIBUTE(nodiscard)
+#define GRPC_MUST_USE_RESULT [[nodiscard]]
+#elif defined(__GNUC__) && !defined(__MINGW32__)
```

Comment:
I don't think this fixes the PHP CI issue, which uses GCC and does not compile as C++ 17 (so no `nodiscard` available). Unless I'm wrong about that, this does not replace https://github.com/grpc/grpc/pull/33702.

Reply:
It doesn't replace it, but it's a path towards doing so that doesn't regress other things.